src/core/meshcop/commissioner.cpp

```diff
@@ -779,6 +779,7 @@ void Commissioner::HandleJoinerFinalize(Coap::Header &aHeader, Message &aMessage
     (void)aMessageInfo;
     StateTlv::State state = StateTlv::kAccept;
     ProvisioningUrlTlv provisioningUrl;
+    uint8_t buf[OPENTHREAD_CONFIG_MESSAGE_BUFFER_SIZE];
```

Comment:
128 bytes on a stack is quite large for embedded systems, especially as it is used only for logging purposes. How about hiding this declaration, and the dumping code itself behind something like:
`#if (OPENTHREAD_ENABLE_CERT_LOG == 1)  ...  #endif`
Or maybe, even better we could provide a Dump method in the message class and use it instead of using extra memory to create a copy?